cmd/protolock/main.go

```diff
@@ -98,16 +105,85 @@ func main() {
 
 	case "status":
 		report, err := protolock.Status(*ignore)
-		if err != nil {
-			handleReport(report, err)
+		if err != protolock.ErrWarningsFound && err != nil {
+			fmt.Println("[protolock]:", err)
+			os.Exit(1)
 		}
 
+		// if plugins are provided, attempt to execute each as a binary located
+		// in the user's OS executable filepath using exec.LookPath
+		if *plugins != "" {
+			report, err = runPlugins(*plugins, report)
+			if err != nil {
+				fmt.Println(err)
+				os.Exit(1)
+			}
+		}
+
+		handleReport(report, err)
+
 	default:
 		os.Exit(0)
 	}
 }
 
-func handleReport(report protolock.Report, err error) {
+func runPlugins(pluginList string, report *protolock.Report) (*protolock.Report, error) {
+	inputData := &bytes.Buffer{}
+
+	err := json.NewEncoder(inputData).Encode(&extend.Data{
+		Current:          report.Current,
+		Updated:          report.Updated,
+		ExistingWarnings: report.Warnings,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	plugins := strings.Split(pluginList, ",")
+	for i, name := range plugins {
+		name = strings.TrimSpace(name)
+		path, err := exec.LookPath(name)
+		if err != nil {
+			return wrapPluginErr(path, err)
```

Comment:
consider using `name` instead, as `path` may be unavailable if an error is returned.